src/shared/sam/localLambdaRunner.ts

```diff
@@ -272,14 +345,113 @@ export async function invokeLambdaFunction(
                 })
             },
         })
-        if (attachResults.success) {
-            await showDebugConsole()
+            .then(r => {
+                if (r.success) {
+                    showDebugConsole()
+                }
+            })
+            .catch(e => {
+                getLogger().error(`Failed to debug: ${e}`)
+                ctx.chanLogger.channel.appendLine(`Failed to debug: ${e}`)
+            })
+
+        if (config.invokeTarget.target !== 'api') {
+            await tryAttach
+        } else {
+            const payload = JSON.parse(await readFile(config.eventPayloadFile, { encoding: 'utf-8' }))
+            await requestLocalApi(ctx, config.api!, config.apiPort!, payload).finally(() => {
+                stopApi(config)
+            })
+            const timer2 = createInvokeTimer(ctx.settings)
+            await waitForPort(config.apiPort!, timer2, ctx.chanLogger, false)
+            await tryAttach
         }
     }
 
     return config
 }
 
+function stopApi(config: SamLaunchRequestArgs) {
+    if (!samStartApi) {
+        // This is a bug, should never happen.
+        getLogger().error('SAM: unknown debug session: %s', config.name)
+        return
+    }
+
+    try {
+        getLogger().verbose('SAM: sending SIGHUP to sam process: pid %d', samStartApi.pid())
+        samStartApi.stop(true, 'SIGHUP')
+    } catch (e) {
+        getLogger().warn('SAM: failed to stop sam process: pid %d: %O', samStartApi.pid(), e as Error)
+    } finally {
+        samStartApi = undefined
+    }
+}
+
+/**
+ * Sends an HTTP request to the local API webserver, which invokes the backing
+ * Lambda, which will then enter debugging.
+ */
+function requestLocalApi(ctx: ExtContext, api: APIGatewayProperties, apiPort: number, payload: any): Promise<void> {
+    return new Promise((resolve, reject) => {
+        const reqMethod = api?.httpMethod?.toUpperCase() ?? 'GET'
+        let reqOpts = {
+            // Sets body to JSON value and adds Content-type: application/json header.
+            json: true,
+            uri: `http://127.0.0.1:${apiPort}${api?.path}`,
+            method: reqMethod,
+            timeout: 4000,
+            headers: api?.headers,
+            body: payload,
+            qs: api?.querystring,
+            // TODO: api?.stageVariables,
+        }
+        ctx.chanLogger.info('AWS.sam.localApi.request', `Sending request to local API: ${reqOpts.uri}`)
+
+        async function retryRequest(retries: number, retriesRemaining: number) {
```

Comment:
Do we need this function nested as well? Same feedback as higher up if we do.

Reply:
Why not nest it? There is nothing special about functions.

Reply:
Mainly because we don't use this terminology within another function. Please use an arrow function.